jib-core/src/main/java/com/google/cloud/tools/jib/ncache/CacheStorage.java

```diff
@@ -36,13 +37,22 @@
 interface CacheStorage {
 
   /**
-   * Saves the {@link CacheWrite}.
+   * Saves the {@link UncompressedCacheWrite}.
    *
-   * @param cacheWrite the {@link CacheWrite}
-   * @return the {@link CacheEntry} for the written {@link CacheWrite}
+   * @param uncompressedCacheWrite the {@link UncompressedCacheWrite}
+   * @return the {@link CacheEntry} for the written {@link UncompressedCacheWrite}
    * @throws IOException if an I/O exception occurs
    */
-  CacheEntry write(CacheWrite cacheWrite) throws IOException;
+  CacheEntry write(UncompressedCacheWrite uncompressedCacheWrite) throws IOException;
+
+  /**
+   * Saves a compressed layer {@link Blob}.
+   *
+   * @param compressedLayerBlob the compressed layer {@link Blob}
+   * @return the {@link CacheEntry} for the written layer
+   * @throws IOException if an I/O exception occurs
+   */
+  CacheEntry write(Blob compressedLayerBlob) throws IOException;
```

Comment:
From the class Java "The cache storage engine stores layer data as UncompressedCacheWrites", I assume this method will/should uncompress the given blob, right?

Reply:
Ah right, I need to fix that javadoc.